### DISCUSSION TITLE
On first open, CPU stays spiked until variables are updated

### DISCUSSION THREAD

**Discussion by ghost** [elkowar/eww#619] [Q&A] [answered]
Hi. Sorry if I am using polling variables or run-while incorrectly. As an example, these variables poll every 0s when the slider is visible (showBrightness) and use ~5% CPU when constantly polling. Almost no CPU is used when the widget is not visible. 

```
(defvar showBrightness false)
(defpoll curBrightness :interval "0s" :run-while showBrightness "light | tr -d '.00'")
(defpoll closeBrightnessListener :interval "1s" 
  :run-while showBrightness "python scripts/autoClose.py brightness")
```

However when the window containing this widget is first launched - showBrightness is false, the slider is hidden, therefore neither variable should be polling - it uses 5% CPU as if it were open. Updating showBrightness to true and then back to false instantly drops CPU back to the expected idle level.

**Comment by viandoxdev** [comment, marked as answer]
Tbh the solution is probably to not use 0s defpolls, these are pretty much never a good idea (spawning at least 3 different processes several times per second, commands aren't as fast as calling a function in a programming language). Why do you need it for exactly?

**Comment by ghost** [comment]
Thanks for the pointer - updating the slider variables (volume, brightness) on keypress accomplishes the same smooth scrolling effect.
